float32 newscale
---
bool ret
